src/App.jsx

```diff
@@ -14,33 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_CHAR = 'SORT_BY_CHAR';
+const SORT_BY_LENGTH = 'SORT_BY_LENGTH';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const Goods = ({ goods }) => (
+  <ul>
+    {goods.map(el => (
+      <li data-cy="Good" key={el}>
+        {el}
+      </li>
+    ))}
+  </ul>
+);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+const sortGoods = (goods, sortType = '', isReverse = false) => {
+  if (!sortType && isReverse) return [...goods].reverse();
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortedGoods = [...goods].sort((a, b) => {
+    switch (sortType) {
+      case SORT_BY_CHAR:
+        return isReverse ? b.localeCompare(a) : a.localeCompare(b);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+      case SORT_BY_LENGTH:
+        // return isReverse ? b.length - a.length : a.length - b.length;
+
+        return a.length - b.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReverse && sortType === SORT_BY_LENGTH
+    ? sortedGoods.reverse()
+    : sortedGoods;
+};
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReverse, setReverse] = useState(false);
+
+  const goods = sortGoods(goodsFromServer, sortType, isReverse);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SORT_BY_CHAR,
+          })}
+          onClick={() => setSortType(SORT_BY_CHAR)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortType !== SORT_BY_LENGTH,
+          })}
+          onClick={() => setSortType(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReverse,
+          })}
+          onClick={() => {
+            setReverse(!isReverse);
+          }}
+        >
+          Reverse
+        </button>
+
+        {(sortType !== '' || isReverse) && (
```

Comment:
the same, but more readable
```suggestion
        {(sortType.length || isReverse) && (
```